jplag.cli/src/main/java/de/jplag/CLI.java

```diff
@@ -125,21 +154,27 @@ public JPlagOptions buildOptionsFromArguments(Namespace namespace) {
         addAllMultiValueArgument(NEW_DIRECTORY.getListFrom(namespace), submissionDirectories);
         addAllMultiValueArgument(OLD_DIRECTORY.getListFrom(namespace), oldSubmissionDirectories);
 
-        JPlagOptions options = new JPlagOptions(submissionDirectories, oldSubmissionDirectories, LANGUAGE.getFrom(namespace));
-        options.setBaseCodeSubmissionName(BASE_CODE.getFrom(namespace));
-        options.setVerbosity(Verbosity.fromOption(VERBOSITY.getFrom(namespace)));
-        options.setDebugParser(DEBUG.getFrom(namespace));
-        options.setSubdirectoryName(SUBDIRECTORY.getFrom(namespace));
-        options.setFileSuffixes(fileSuffixes);
-        options.setExclusionFileName(EXCLUDE_FILE.getFrom(namespace));
-        options.setMinimumTokenMatch(MIN_TOKEN_MATCH.getFrom(namespace));
-        options.setSimilarityThreshold(SIMILARITY_THRESHOLD.getFrom(namespace));
-        options.setMaximumNumberOfComparisons(SHOWN_COMPARISONS.getFrom(namespace));
-        ComparisonMode.fromName(COMPARISON_MODE.getFrom(namespace)).ifPresentOrElse(options::setComparisonMode,
-                () -> logger.warn("Unknown comparison mode, using default mode!"));
+        var language = LanguageLoader.getLanguage(LANGUAGE.getFrom(namespace)).orElseThrow();
+        JPlagOptions options = new JPlagOptions(language, submissionDirectories, oldSubmissionDirectories) //
+                .withBaseCodeSubmissionName(BASE_CODE.getFrom(namespace)) //
+                .withVerbosity(Verbosity.fromOption(VERBOSITY.getFrom(namespace))) //
+                .withDebugParser(DEBUG.getFrom(namespace)) //
+                .withSubdirectoryName(SUBDIRECTORY.getFrom(namespace)) //
+                .withFileSuffixes(Arrays.stream(fileSuffixes).toList()) //
+                .withExclusionFileName(EXCLUDE_FILE.getFrom(namespace)) //
+                .withMinimumTokenMatch(MIN_TOKEN_MATCH.getFrom(namespace)) //
+                .withSimilarityThreshold(SIMILARITY_THRESHOLD.getFrom(namespace)) //
+                .withMaximumNumberOfComparisons(SHOWN_COMPARISONS.getFrom(namespace));
```

Comment:
For brevity (and also to avoid constructing n new option instances) we could also use the full constructor here, right?

Reply:
depends. It may be the case that we do not want to set all options via CLI. (or we may add more to JPlagOptions later)

Reply:
I would say it makes the code more compact (and marginally more performant) and we don't use readability as the arguments are identified anyways by their `<Name.getFrom()` syntax.